src/framework/MOM_document.F90

```diff
@@ -637,19 +637,33 @@ function real_string(val)
   elseif (val == 0.) then
     real_string = "0.0"
   else
-    if ((abs(val) <= 1.0e-100) .or. (abs(val) >= 1.0e100)) then
-      write(real_string(1:32), '(ES24.14E3)') val
-      if (.not.testFormattedFloatIsReal(real_string,val)) &
-        write(real_string(1:32), '(ES24.15E3)') val
+    if ((abs(val) < 1.0e-99) .or. (abs(val) >= 1.0e100)) then
+      write(real_string(1:32), '(ES24.14E4)') val
+      if (scan(real_string, "eE") == 0) then  ! Fix a bug with a missing E in PGI formatting
```

Comment:
Perhaps we should at least acknowledge that PGI is the only one following the standard here?  Technically, it is Intel and GNU which have the bug.

Reply:
Also, I believe we can use `E3` with all three compilers and it will produce three digits of output, which would eliminate the need to remove leading zeros.

Reply:
Skipping writing the E is not a part of the standard, is it? The numbers written without the E make no sense.  So for the question of adding the E, it is gnu and intel that are correct, and PGI has a bug, doesn't it?
As I understand it, on the question of whether we should be using E3 or E4 to get the right result with 3 digits in the exponent, pgi is doing the right thing.  On the other hand, if the standard says use E3 to get 3 digits (not counting the sign of the exponent), I would be happy changing this to E3.

Reply:
I misread the standard, you are correct.  The `E` in the output can only be omitted when the `E` is not in the format string (e.g. `E24.14`).  If the E is included (`E24.14E3`) then the `E` is supposed to appear in the number.  So PGI is not conformant in this case.  My mistake!

Reply:
So should we take the (demonstrably working if perhaps inelegant) code change as is, or are there changes that you would like to to suggest?

Reply:
yes current form is good